packages/react-ui/src/TableOfContent/useTOCReducer.js

```diff
@@ -0,0 +1,32 @@
+import { useReducer, useCallback } from "react";
```

Comment:
I find it clearer if this hooks has it own a dedicated test

Reply:
sure 👌